glue/core/hub.py

```diff
@@ -175,9 +184,13 @@ def _find_handlers(self, message):
             # narrow to the most-specific message
             candidate = max(messages, key=_mro_count)
 
-            handler, test = subscriptions[candidate]
+            handler, test, priority = subscriptions[candidate]
             if test(message):
-                yield subscriber, handler
+                priorities.append(priority)
+                prioritized_handlers.append((subscriber, handler))
+
+            for i in np.argsort(priorities):
+                yield prioritized_handlers[i]
```

Comment:
This would then be something like:

```
for subscriber, handler, _ in sort(prioritized_handlers, key=x[2], reversed=True):
    yield subscriber, handler
```